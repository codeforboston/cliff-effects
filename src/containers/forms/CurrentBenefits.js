import { connect } from 'react-redux';

import { CurrentBenefitsStep } from '../../forms/CurrentBenefits';
import { setHasBenefit } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),

    USState: state.getIn([
      'geography',
      'state',
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setHasBenefit({ benefit, value }) {
      dispatch(
        setHasBenefit({
          time: 'current',
          benefit,
          value,
        })
      );
    },
  };
};

const CurrentBenefitsContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentBenefitsStep);

export { CurrentBenefitsContainer as CurrentBenefitsStep };
