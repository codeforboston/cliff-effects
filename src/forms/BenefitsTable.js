// REACT COMPONENTS
import React from 'react';
import {
  Table
} from 'semantic-ui-react';

// CUSTOM COMPONENTS
// Both the table and graph should just be added to a results page, but
// this will do for now

// BENEFIT LOGIC
import { getSNAPBenefits } from '../programs/federal/snap';
import { getHousingBenefit } from '../programs/state/massachusetts/housing';

// OBJECT MANIPULATION
import { cloneDeep } from 'lodash';


const getSignSymbol = function ( num ) {
  if ( num > 0 ) { return '+'; }
  else if ( num < 0 ) { return '-'; }
  else { return ''; }
};  // End getSignSymbol()


const BenefitsTable = function ( props ) {

  var currentClient = cloneDeep( props.client ),
      futureClient  = cloneDeep( currentClient );
  // Hack because everything's using future value instead
  // of just using current value.
  currentClient.future.earned = currentClient.current.earned;




  var SNAPBenefitCurrent  = Math.round( getSNAPBenefits( currentClient ).benefitValue * 12 ),
      SNAPBenefitFuture   = Math.round( getSNAPBenefits( futureClient ).benefitValue * 12 ),
      SNAPDiff            = SNAPBenefitFuture - SNAPBenefitCurrent,
      sec8BenefitCurrent  = Math.round( getHousingBenefit( currentClient ).benefitValue * 12 ),
      sec8BenefitFuture   = Math.round( getHousingBenefit( futureClient ).benefitValue * 12 ),
      sec8Diff            = sec8BenefitFuture - sec8BenefitCurrent,
      totalBenefitCurrent = SNAPBenefitCurrent + sec8BenefitCurrent,
      totalBenefitFuture  = SNAPBenefitFuture + sec8BenefitFuture,
      totalDiff           = SNAPDiff + sec8Diff,
      incomeCurrent       = Math.round( currentClient.current.earned * 12 ),
      incomeFuture        = Math.round( futureClient.future.earned * 12 ),
      incomeDiff          = incomeFuture - incomeCurrent,
      netCurrent          = totalBenefitCurrent + incomeCurrent,
      netFuture           = totalBenefitFuture + incomeFuture,
      netDiff             = totalDiff + incomeDiff;


const   columnHeaderStyle = {
                            background: 'rgba(0, 181, 173, 1)',
                            color: 'white',
                            fontSize: '1.3em',
                            fontWeight: 900,
                            textAlign: 'center',
                            borderRadius: 'inherit',
                            letterSpacing: '0.02em',
                            }
      , totalsRowStyle    = {
                            borderTop: "2px solid rgba(0, 181, 173, 1)",
                            fontWeight: 700,
                            fontSize: '1.1em',
                            padingTop: '0.25em'
                              }
      , rowHeaderStyle    = {
                    fontSize: '1.1em',
                    fontWeight: 700,
                    textAlign: 'left'
                    }
      , totalsRowHeaderStyle = {
                    fontSize: '1.2em',
                    fontWeight: 700,
                    textAlign: 'left',
                    borderTop: "2px solid rgba(0, 181, 173, 1)",
                    padingTop: '0.25em'


      };






const SNAPBenefitRow = function( props ){
  if(!currentClient.current.hasSnap ) return (null)
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>SNAP</Table.Cell>
        <Table.Cell textAlign='right'>${SNAPBenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${SNAPBenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(SNAPDiff) } ${Math.abs(SNAPDiff)} / year</Table.Cell>
      </Table.Row>
    )
  };

const Sec8BenefitRow  = function( props ){  
  if(!currentClient.current.hasHousing) return (null)
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>Section 8 Housing</Table.Cell>
        <Table.Cell textAlign='right'>${sec8BenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${sec8BenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(sec8Diff) } ${Math.abs(sec8Diff)} / year</Table.Cell>
      </Table.Row>
    ) 
};

const TotalBenefitsRow = function( props ){
  if(!currentClient.current.hasSnap || !currentClient.current.hasHousing) return (null)
    return(
      <Table.Row>
        <Table.Cell textAlign='right' width={3} style={totalsRowHeaderStyle}>Total Benefits</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${totalBenefitCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${totalBenefitFuture} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>{ getSignSymbol(totalDiff) } ${Math.abs(totalDiff)} / year</Table.Cell>
      </Table.Row>
    )
};

const IncomeRow = function ( props ) { 
    return (
      <Table.Row>
        <Table.Cell style={rowHeaderStyle}>Income</Table.Cell>
        <Table.Cell textAlign='right'>${incomeCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right'>${incomeFuture} / year</Table.Cell>
        <Table.Cell textAlign='right'>{ getSignSymbol(incomeDiff) } ${Math.abs(incomeDiff)} / year</Table.Cell>
      </Table.Row>
    )
};

const TotalsRow = function ( props ) {
    return (
      <Table.Row style={{border: 'none'}}>
        <Table.Cell textAlign='right' width={3} style={totalsRowHeaderStyle}>Net Total</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${netCurrent} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>${netFuture} / year</Table.Cell>
        <Table.Cell textAlign='right' width={3} style={totalsRowStyle}>{ getSignSymbol(netDiff) } ${Math.abs(netDiff)} / year</Table.Cell>
      </Table.Row>
    )
  };


  return (
    <wrapper>
      <Table celled>
       <Table.Header>
          <Table.Row >
            <Table.Cell style={columnHeaderStyle} width={3}>Benefit</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>Current Benefits</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>New Estimate</Table.Cell>
            <Table.Cell style={columnHeaderStyle} width={3}>Difference</Table.Cell>
          </Table.Row>
      </Table.Header>
        <Table.Body>
          <SNAPBenefitRow client={currentClient} />
          <Sec8BenefitRow client={currentClient} />
          <TotalBenefitsRow client={currentClient} />
          <IncomeRow />
          <TotalsRow />
        </Table.Body>
      </Table>
    </wrapper>
  )

};  // End BenefitsTable(<>)


export {
	BenefitsTable
};
