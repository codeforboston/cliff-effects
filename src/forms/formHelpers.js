// REACT COMPONENTS
import React, { Component } from 'react';
import {
  // Generic Form stuff
  Button,
  Header,
  Segment,
  Divider,
  Form,
  Grid,
  // Input,
  Checkbox,
  // Money columns
} from 'semantic-ui-react';

// UTILITIES
import { toMonthlyAmount } from '../utils/math';
import { isPositiveNumber } from '../utils/validators';
import { toMoneyStr } from '../utils/prettifiers';


// ========================================
// GENERIC COMPONENTS
// ========================================

/** Returns a component with a massive teal button
 * 
 */
const MassiveButton = function ({ className, func, children }) {

  className = (className || '') + ' massive-button';
  return (
    <Button fluid type='button' color='teal' size='large' className={className} onClick={func}>
      { children }
    </Button>
  );

};  // End MassiveButton(<>)

/**
 * Link that opens new tab
 */
const ExternalLink = function ({ href, children }) {
  return (<a href={href} target='_blank'>{children}</a>);
};


/** Returns a Grid Column containing a button of the style used
* to navigate backwards and forwards through steps of the form.
*
* @function
* @param {object} props
* @property {object} props.func - A function that is run when the
* button is clicked.
* @property {string} props.name - The text to be displayed on
* the button.
*
* @returns Component
*/
const BottomButton = function(props){
  return (
    <Grid.Column className={'large-bottom-button'} width={3}>
      <MassiveButton {...props} />
    </Grid.Column>
  );
};  // End BottomButton() Component

/** The row containing the 'Previous' and 'Next' buttons at the
* bottom of each form page.
*
* @function
* @param {object} propsContainer - Containing the `props` object in
* another object seemed to be the only way to pass on `props`.
* @property {object} propsContainer.props - Properties needed
* for the functional aspects of the Component. @todo Move the
* functions `previousStep()` and `nextStep()` into here if possible.
* @property {object} propsContainer.props.previousStep() - Function
* that goes back to the previous form element.
* @property {object} propsContainer.props.nextStep() - Function
* that goes forward to the next form element.
*
* @todo Use this somehow for the first page and last page too,
* but then hide the 'previous' and the 'next' respectively? Or
* split those into their own Components?
*
* @returns Component
*/
const BottomButtons = function({ left, right }) {
  const flexItemStyle = { flexBasis: '118.3px' };
  const buttonProps = { style: flexItemStyle, type: 'button', color: 'teal', size: 'large' }; 
  return (
    <div style={{ display: 'flex', justifyContent: 'space-between' }}>
      { left ?
        <Button {...buttonProps} onClick={left.func}>
          { left.name }
        </Button>
        :
        <div style={flexItemStyle} />
      }
      { right ?
        <Button {...buttonProps} onClick={right.func}>
          { right.name }
        </Button>
        :
        <div style={flexItemStyle} />
      }
    </div>
  );
};  // End BottomButtons() Component


/** Constructor for all the stuff that's supposed to go inside
* the Form Component. Does not include the `<Form>` Component
* as a container because it looks like that needs to be unique
* (the 'CurrentBenefitsStep' gives it `size='massive'`).
*
* @function
* @param {object} props
* @property {string} props.title - Text to go in the `h1` element.
* @property {string} props.clarifier - Text to go in the `h3`
* element, giving some description, instructions, or clarifications.
* @property {string} props.children - Component(s) to be inserted
* into the middle - a custom form section containing inputs, etc.
* @property {Object} props.next - the 'next form section' function
* @property {Object} props.prev - the 'previous form section' function
*
* @returns Component
*/
const FormPartsContainer = function(props) {
  return (
    <Segment padded='very' style={{ minHeight: '600px' }}>
      <Segment style={{ minHeight: '500px' }} basic={true}>
        <Header as='h1' color='teal' textAlign='center'>
          { props.title }
        </Header>
        { !props.clarifier
          ? null
          : <Header as='h3' textAlign='center'>
              { props.clarifier }
            </Header>
        }

        { props.children }

      </Segment>
      <Divider />
      <BottomButtons left={props.left} right={props.right} next={props.next} prev={props.prev} />

    </Segment>
  );
};  // End FormPartsContainer() Component


/** Toggle with size='massive'
*
* @function
* @param {object} props - sent from attributes of parent element
* @property {string} props.label - Text displayed next to toggle
* @property {string} props.value - Should reflect state change and
* will be displayed as the value. Flow goes:
* user input > state > value > visual feedback
* @property {object} props.setClientProperty - Function that changes page
* state (changes the "source of truth")
* @property {string} props.name - The key of the client property that
* will be saved and will be used to fill in the field's value.
*
* @returns Component
*/
const MassiveToggle = function (props) {

  /** @todo Switch props.setClientProperty to props.onChange everywhere */
  return (
    <Form.Field className='massive-toggle'
      toggle
      label={ props.label }
      checked={ props.checked || props.value }
      onChange={ props.onChange || props.setClientProperty }
      name={ props.name || props.name }
      control={ Checkbox }
      size='massive'/>
  );

};  // End MassiveToggle{} Component


/** A clearer way than a ternary operator to have a possible
* subheader. Also, clearer for separate styling
*
* @function
* @param {object} props
* @property {object} props.children - Contents of this element
*
* @returns Component
*/
const FormSubheading = function ( props ) {

  if ( !props.children ) { return null; }

  return (
    <wrapper className = { 'form-subheading' }
      style={{ display: 'block', textAlign: 'left' }}>
        { props.children }
    </wrapper>
  );

};  // End FormSubheading{} Component


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const FormHeading = function ({ subheading, children }) {

  if ( !children ) { return null; }

  return (
    <wrapper className={'form-heading'} >
      <div></div> {/** div here to make sure header margin doesn\'t collapse */}
      <Header as='h3' style={{ display: 'inline-block' }}>
        { children }
      </Header>
      <FormSubheading>{subheading}</FormSubheading>
      <br/>
    </wrapper>
  );

};  // End FormHeading{} Component


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const InlineLabelInfo = function ( props ) {

  var labelInfoDisplay = ' hidden'; // Will be '' in future

  // return (
  //   <wrapper className = { 'label-info info-revealer' + labelInfoDisplay } style = {{
  //     position: 'relative', top: '-0.5em',
  //     marginLeft: '1em', padding: '0.1em 0.2em',
  //     textAlign: 'left', verticalAlign: 'middle',
  //     border: '1px solid black'
  //   }}>
  //     <wrapper className='info-indicator'>i</wrapper>
  //     <wrapper className='info-tooltip'>{props.children}</wrapper>
  //   </wrapper>
  // );

  return (
    <wrapper className = { 'label-info' + labelInfoDisplay }
      style = {{ marginLeft: '1em' }}>
        { props.children }
    </wrapper>
  );

};  // End InlineLabelInfo{} Component

// Possible tooltip version of labels:
// (could be made official in the Row creator with conditionals)
// <label>Earned Income
//   <style type='display on hover handled in css'></style>
//   <div
//     className={ 'info-revealer' }
//     style={{
//       position: 'relative',
//       display: 'inline-block',
//       fontSize: '10px',
//       border: '1px solid black',
//       margin: '1em',
//       top: '-0.5em',
//       textAlign: 'center',
//       width: '1.6em',
//       height: '1.6em'
//     }}>
//     <div style={{ position: 'relative', top: '-0.2em' }}>i</div>
//     <div
//       className={ 'info-tooltip' }
//       style={{ position: 'absolute', padding: '.2em' }}
//     >
//       Weekly income = hourly wage times average number of work hours per week
//     </div>
//   </div>
// </label>



// ========================================
// MONEY ON INTERVALS COLUMNS COMPONENTS
// ========================================

// Ideas of how to handle a different styling situation (if the designers switch columns)

// If we want more control over placement, we may look into this:
// <Grid textAlign='center' verticalAlign='middle'>
//   <Grid.Row className='inputs-in-right-column'>
//     <Grid.Column className='left-label'>
//       <label>Earned Income</label>
//     </Grid.Column>
//     <Grid.Column className='right-input'>
//       <Input type='number'/>
//     </Grid.Column>
//   </Grid.Row>
// </Grid>

// <Form.Field inline>
//   <span className='column-1-header'>Income Source</span>
//   <div className='right-column'>
//     <span className='Weekly'>Income Source</span>
//     <span className='Monthly'>Income Source</span>
//     <span className='Yearly'>Income Source</span>
//   </div>
//   <Input
//     type='number'
//     onChange={props.setClientProperty}
//     className='right-column'
//     name='Earned Income' placeholder='Earned Income'
//   />


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const ColumnHeading = function ({ type, colName, style, children }) {
  var classes = type + '-column cashflow-column header ' + colName;
  return (
    <Header as='h4' className={classes} style={style} color='teal'>{children}</Header>
  );
};  // End ColumnHeading()


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const IntervalColumnHeadings = function ({ type }) {

  var columnTitle = type.toLowerCase().replace(/\b[a-z]/g, letter => letter.toUpperCase()) + " Type",
      styles      = { fontSize: '14px' };

  return (
    <wrapper style={{ display: 'inline-block' }}>
      <ColumnHeading type={type} colName='weekly'  style={styles}>Weekly</ColumnHeading>
      <ColumnHeading type={type} colName='monthly' style={styles}>Monthly</ColumnHeading>
      <ColumnHeading type={type} colName='yearly'  style={styles}>Yearly</ColumnHeading>
      <ColumnHeading type={type} colName={type} style={styles} columnTitle={columnTitle}>{columnTitle}</ColumnHeading>
    </wrapper>
  );

};  // End IntervalColumnHeadings{} Component


/** 
 * @todo description
 * @todo Write callback descriptions for function params: http://usejsdoc.org/tags-callback.html
 * 
 * @param {Object} props
 * @param {number || string} props.value - Valid client value
 * @param {string} props.name - For HTML name property
 * @param {string} props.className - HTML class names
 * @param {*} [props.otherData] - Sent back to `store()`
 * @param {function} props.format - Given `value`. Must return what you want shown in the number field.
 * @param {function} props.validate - Given `value`. Must return boolean.
 * @param {function} props.store - Given an event, `value`, [`otherData`]
 */
class ManagedNumberField extends Component {
  constructor ( props ) {
    super( props );
    // Need updating
    this.state = {
      focused:    false,
      valid:      true,
      focusedVal: this.props.value,
    };
  }  // End constructor()
  
  handleFocus = ( evnt, inputProps ) => {
    var newState = {
      focused:    true,
      // `value` here is probably a long decimal
      focusedVal: this.props.format( this.props.value )
    }
    this.setState( newState );
  }
  
  handleBlur = ( evnt ) => {
    this.setState({ focused: false, valid: true });
  }

  handleChange = ( evnt, inputProps ) => {
    var value = inputProps.value,
        valid = this.props.validate( value );

    if ( valid ) {
      this.props.store( evnt, inputProps, this.props.otherData );
    }

    this.setState({ valid: valid, focusedVal: value });
  }  // End handleChange()

  render() {

    var { valid, focused, focusedVal }  = this.state;
    var { value, name, className }      = this.props;

    // Format correctly when neighbors are updated, if needed
    if ( !focused ) { value = this.props.format( value ) }
    else { value = focusedVal; }

    /** @todo Different class for something 'future' that has a current value that isn't 0 */
    return (
      <Form.Input
        error     = { !valid }
        value     = { value }
        name      = { name }
        className = { className }
        onChange  = { this.handleChange }
        onFocus   = { this.handleFocus }
        onBlur    = { this.handleBlur }
        type      = { 'number' } />
    );

  }  // End render()

};  // End ManagedNumberField


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const CashFlowRow = function ({ generic, timeState, setClientProperty, children, labelInfo, type, time }) {

  var updateClient = function ( evnt, inputProps, interval ) {
    var monthly = toMonthlyAmount[ interval ]( evnt, inputProps.value ),
        obj     = { name: generic, value: monthly };
    setClientProperty( evnt, obj );
  };

  /** baseVal
   * Get the time ('future' or 'current') monthly value unless there is
   *     none, in which case, get the 'current' monthly cash flow value
   *     (to prefill future values with 'current' ones if needed).
   *
   * @var
   *
   * @todo Add some kind of UI indication when it's the same as the 'current'
   *     value. What if some of the row's values are the same and some are
   *     different?
   */
  var baseVal   = timeState[ generic ],
      classes   = [ time, type, 'cashflow-column' ],
      baseProps = {
        store:    updateClient,
        validate: isPositiveNumber,
        format:   toMoneyStr
      };

  return (
    <Form.Field inline className={'cashflow'}>
      <ManagedNumberField
        {...baseProps}
        value     = { baseVal / 4.33 }
        name      = { generic + 'Weekly' }
        className = { classes.concat( 'weekly' ).join(' ') }
        otherData = { 'weekly' }
      />
      <ManagedNumberField
        {...baseProps}
        value     = { baseVal }
        name      = { generic }
        className = { classes.concat( 'monthly' ).join(' ') }
        otherData = { 'monthly' }
      />
      <ManagedNumberField
        {...baseProps}
        value     = { baseVal * 12 }
        name      = { generic + 'Yearly' }
        className = { classes.concat( 'yearly' ).join(' ') }
        otherData = { 'yearly' }
      />
      <wrapper className={'cashflow-column'}>
        <label>{ children }</label>
        <InlineLabelInfo>{ labelInfo }</InlineLabelInfo>
      </wrapper>
    </Form.Field>
  );

};  // End CashFlowRow{} Component


/** @todo Separate into different files? */
export {
  ExternalLink,
  BottomButtons, FormPartsContainer, BottomButton,
  MassiveToggle, FormSubheading, FormHeading,
  InlineLabelInfo,
  IntervalColumnHeadings, ColumnHeading, ManagedNumberField,
  CashFlowRow
};
