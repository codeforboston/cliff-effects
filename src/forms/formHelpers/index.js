import BottomButtons from './BottomButtons';
import FormPartsContainer from './FormPartsContainer';
import BottomButton from './BottomButton';
import MassiveToggle from './MassiveToggle';
import FormSubheading from './FormSubheading';
import FormHeading from './FormHeading';
import InlineLabelInfo from './InlineLabelInfo';
import IntervalColumnHeadings from './IntervalColumnHeadings';
import ColumnHeading from './ColumnHeading';
import CashFlowRow from './CashFlowRow';

export {
  BottomButtons,
  FormPartsContainer,
  BottomButton,
  MassiveToggle,
  FormSubheading,
  FormHeading,
  InlineLabelInfo,
  IntervalColumnHeadings,
  ColumnHeading,
  CashFlowRow
};

// ========================================
// MONEY ON INTERVALS COLUMNS COMPONENTS
// ========================================

// Ideas of how to handle a different styling situation (if the designers switch columns)

// If we want more control over placement, we may look into this:
// <Grid textAlign='center' verticalAlign='middle'>
//   <Grid.Row className='inputs-in-right-column'>
//     <Grid.Column className='left-label'>
//       <label>Earned Income</label>
//     </Grid.Column>
//     <Grid.Column className='right-input'>
//       <Input type='number'/>
//     </Grid.Column>
//   </Grid.Row>
// </Grid>

// <Form.Field inline>
//   <span className='column-1-header'>Income Source</span>
//   <div className='right-column'>
//     <span className='Weekly'>Income Source</span>
//     <span className='Monthly'>Income Source</span>
//     <span className='Yearly'>Income Source</span>
//   </div>
//   <Input
//     type='number'
//     onChange={props.setClientProperty}
//     className='right-column'
//     name='Earned Income' placeholder='Earned Income'
//   />
