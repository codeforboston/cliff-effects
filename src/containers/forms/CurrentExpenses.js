import { connect } from 'react-redux';

import { CurrentExpensesStep } from '../../forms/CurrentExpenses';

import { setCashValue, setHousingType, setPaysUtility, setGetsFuelAssistance } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setExpenseValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'current',
          name,
          value,
        })
      );
    },

    setHousingType({ housingType }) {
      dispatch(
        setHousingType({
          time: 'current',
          housingType,
        })
      );
    },

    setPaysUtility({ utility, paysUtility }) {
      dispatch(
        setPaysUtility({
          time: 'current',
          utility,
          paysUtility,
        })
      );
    },

    setGetsFuelAssistance({ getsAssistance }) {
      dispatch(
        setGetsFuelAssistance({
          time: 'current',
          getsAssistance,
        })
      );
    },
  };
};

const CurrentExpensesStepContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentExpensesStep);

export { CurrentExpensesStepContainer as CurrentExpensesStep };
