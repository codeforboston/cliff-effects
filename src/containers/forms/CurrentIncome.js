import { connect } from 'react-redux';

import { CurrentIncomeStep } from '../../forms/CurrentIncome';

import { setCashValue } from '../../actions';

const mapStateToProps = (state) => {
  return {
    currentClient: state.getIn([
      'client',
      'current', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setIncomeValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'current',
          name,
          value,
        })
      );
    },
  };
};

const CurrentIncomeStepContainer = connect(mapStateToProps, mapDispatchToProps)(CurrentIncomeStep);

export { CurrentIncomeStepContainer as CurrentIncomeStep };
