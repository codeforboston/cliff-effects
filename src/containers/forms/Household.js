import { connect } from 'react-redux';

import { HouseholdStep } from '../../forms/Household';
import {
  addMember,
  removeMember,
  setMemberIsDisabled,
  setMemberRole,
  setMemberAge,
} from '../../actions';

const time = 'current';

const mapStateToProps = (state) => {
  return {
    household: state.getIn([
      'client',
      'current',
      'household', 
    ]),
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setMemberAge({ index, age }) {
      dispatch(
        setMemberAge({
          time,
          index,
          age,
        })
      );
    },

    setMemberIsDisabled({ index, isDisabled }) {
      dispatch(
        setMemberIsDisabled({
          time,
          index,
          isDisabled,
        })
      );
    },

    setMemberRole({ index, role }) {
      dispatch(
        setMemberRole({
          time,
          index,
          role,
        })
      );
    },

    removeMember({ index }) {
      dispatch(
        removeMember({
          time,
          index,
        })
      );
    },

    addMember({ member }) {
      dispatch(
        addMember({
          time,
          member,
        })
      );
    },
  };
};

const HouseholdStepContainer = connect(mapStateToProps, mapDispatchToProps)(HouseholdStep);

export { HouseholdStepContainer as HouseholdStep };
