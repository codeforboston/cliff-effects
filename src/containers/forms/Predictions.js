import { connect } from 'react-redux';

import { PredictionsStep } from '../../forms/Predictions';

import { setCashValue } from '../../actions';

const mapStateToProps = (state) => {
  return { client: state.get('client') };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setPredictionValue({ name, value }) {
      dispatch(
        setCashValue({
          time: 'future',
          name,
          value,
        })
      );
    },
  };
};

const PredictionsStepContainer = connect(mapStateToProps, mapDispatchToProps)(PredictionsStep);

export { PredictionsStepContainer as PredictionsStep };
