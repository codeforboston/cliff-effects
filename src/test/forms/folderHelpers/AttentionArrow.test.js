import React from 'react';
import { mount } from 'enzyme';

import { AttentionArrow } from '../../../forms/formHelpers';

describe('<AttentionArrow>', () => {
  it('matches snapshot', () => {
    expect(mount(<AttentionArrow />)).toMatchSnapshot();
  });
});
