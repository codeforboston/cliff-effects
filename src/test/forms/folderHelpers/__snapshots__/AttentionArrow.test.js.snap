// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AttentionArrow> matches snapshot 1`] = `
<AttentionArrow>
  <span
    className="attention-arrow"
  >
    <Icon
      as="i"
      className="attention-font"
      fitted={true}
      name="angle right"
      size="big"
    >
      <i
        aria-hidden="true"
        className="angle right big fitted icon attention-font"
      />
    </Icon>
    <Icon
      as="i"
      className="attention-font"
      fitted={true}
      name="angle right"
      size="big"
    >
      <i
        aria-hidden="true"
        className="angle right big fitted icon attention-font"
      />
    </Icon>
  </span>
</AttentionArrow>
`;
